tcms/telemetry/api.py

```diff
@@ -169,3 +169,46 @@ def _append_status_counts_to_result(count, result):
     for status in TestExecutionStatus.chart_status_names:
         status_count = count.get(status, 0)
         result.get(status).append(status_count)
+
+
+@rpc_method(name='Testing.test_case_health')
+def test_case_health(query=None):
+
+    if query is None:
+        query = {}
+
+    all_test_executions = TestExecution.objects.filter(**query)
+
+    test_executions = _get_count_for(all_test_executions)
+    passed_test_executions = _get_count_for(all_test_executions,
+                                            TestExecutionStatus.passed_status_names)
+    failed_test_executions = _get_count_for(all_test_executions,
+                                            TestExecutionStatus.failure_status_names)
+
+    data = {}
+    for te in test_executions:
+        data[te['case_id']] = {
+            'case_id': te['case_id'],
+            'case_summary': te['case__summary'],
+            'count': {
+                'all': te['count'],
+                'success': 0,
+                'fail': 0
+            }
+        }
+
+    for te in passed_test_executions:
+        data[te['case_id']]['count']['success'] = te['count']
+    for te in failed_test_executions:
+        data[te['case_id']]['count']['fail'] = te['count']
+
+    return [v for v in data.values()]
```

Comment:
`list(data.values())` should also work. Returning `data.values()` may also work, I don't know if modernrpc will figure it out and convert that to JSON automatically. You can give it a try via the JS client in the browser.

Reply:
`data.values()` - `"Internal error: Unable to serialize result as valid JSON: Object of type 'dict_values' is not JSON serializable"` , nope, not able to serialize that

Reply:
`list(data.values())` ?